dashboard/app/models/sections/section.rb

```diff
@@ -282,6 +282,34 @@ def toggle_hidden_script(script, should_hide)
     end
   end
 
+  # One of the contstraints for teachers looking for discount codes is that they
+  # have a section in which 10+ students have made progress on 5+ levels in both
+  # csd2 and csd3
+  def has_sufficient_discount_code_progress?
+    return false if students.length < 10
+    csd2 = Script.get_from_cache('csd2')
+    csd3 = Script.get_from_cache('csd3')
+    return false unless csd2 && csd3
+
+    csd2_programming_level_ids = csd2.levels.select {|level| level.is_a?(Gamelab)}.map(&:id)
+    csd3_programming_level_ids = csd3.levels.select {|level| level.is_a?(Weblab)}.map(&:id)
```

Comment:
You've got these backwards: CSD2 is Weblab and CSD3 is Gamelab.

Reply:
I could have sworn I tested these queries in the console, but you are right. Thanks for catching this!